avro/src/test/java/com/fasterxml/jackson/dataformat/avro/interop/ApacheAvroInteropUtil.java

```diff
@@ -108,7 +108,7 @@ protected Schema createSchema(Type type, Map<String, Schema> names) {
                     }
                 }
             }
-            if (type instanceof Class<?> && ((Class<?>) type).getSuperclass() != null) {
+            if (type instanceof Class<?> && ((Class<?>) type).getSuperclass() != null && !Enum.class.isAssignableFrom((Class<?>) type)) {
```

Comment:
Yes. Atlhough the self-recursive type definitions exist elsewhere too, so same problem could occur with builder-style things. But enums may be the only common example.